src/examples/src/widgets/form/Basic.tsx

```diff
@@ -0,0 +1,165 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import { icache } from '@dojo/framework/core/middleware/icache';
+
+import Button from '@dojo/widgets/button';
+import TextInput from '@dojo/widgets/text-input';
+import Form from '@dojo/widgets/form';
+
+const factory = create({ icache });
+
+interface Fields {
+	firstName: string;
+	middleName?: string;
+	lastName: string;
+	email?: string;
+}
+
+const App = factory(function({ middleware: { icache } }) {
+	const results = icache.get<Partial<Fields>>('basic');
+	const onValueResults = icache.get<Partial<Fields>>('basicOnValue');
+
+	return (
+		<virtual>
+			<Form
+				initialValue={{
+					firstName: 'Billy'
+				}}
+				onSubmit={(values) => icache.set('basic', values)}
```

Comment:
do you think this needs both an `onSubmit` and an `onValue` ?

Reply:
I would imagine we need some kind of onSubmit call back that can be hooked up to the form. Passing the values as part of that just seems natural.